src/mpid/ch4/generic/am/mpidig_am_part_callbacks.c

```diff
@@ -144,8 +144,11 @@ int MPIDIG_part_send_data_target_msg_cb(int handler_id, void *am_hdr, void *data
     MPIR_Request *part_rreq = msg_hdr->rreq_ptr;
     MPIR_Assert(part_rreq);
 
-    /* Erroneous if received data with a non-CTS rreq */
-    MPIDIG_PART_CHECK_RREQ_CTS(part_rreq);
+#ifdef HAVE_ERROR_CHECKING
+    /* If this happens, fix the code */
+    MPIR_Assert(MPL_atomic_load_int(&MPIDIG_PART_REQUEST(part_rreq, status)) ==
+                MPIDIG_PART_STATUS_CAN_CTS);
+#endif
 
```

Comment:
The error handling change itself is OK, but better to move to a separate commit.